File: algorithms/cpp/55.跳跃游戏.cpp
#include <iostream>
#include <vector>

using namespace std;
/*
 * @lc app=leetcode.cn id=55 lang=cpp
 *
 * [55] 跳跃游戏
 */

// @lc code=start
class Solution {
 public:
  bool canJump(vector<int>& nums) {
    int last = nums.size() - 1;
    while (last) {
      int can_arrived = false;
      for (int i = last - 1; i > -1; i--) {
        if (i + nums[i] >= last) {
          last = i;
          can_arrived = true;
          break;
        }
      }
      if (!can_arrived) return false;
    }
    return true;
  }
};
// @lc code=end

File: algorithms/js/55.跳跃游戏.js
/*
 * @lc app=leetcode.cn id=55 lang=javascript
 *
 * [55] 跳跃游戏
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
  let last = nums.length - 1;
  while (last) {
    let canArrived = false;
    for (let i = last - 1; i >= 0; i--) {
      if (i + nums[i] >= last) {
        last = i;
        canArrived = true;
        break;
      }
    }

    if (!canArrived) {
      return false;
    }
  }
  return true;
};
// @lc code=end
